internal/layer2/announcer_test.go

```diff
@@ -27,6 +28,8 @@ func Test_SetBalancer_AddsToAnnouncedServices(t *testing.T) {
 
 	for _, service := range services {
 		announce.SetBalancer(service.name, service.ip)
+		//we need to empty spamCh
```

Comment:
```suggestion
		// We need to empty spamCh.
```

If you comment here I'd explain *why* we need to empty the channel.

Reply:
Changed:
// We need to empty spamCh as spamLoop() is not started.